providers/rocketflag/pkg/provider.go

```diff
@@ -0,0 +1,120 @@
+package rocketflag
+
+import (
+	"context"
+
+	"github.com/open-feature/go-sdk/openfeature"
+	rocketflag "github.com/rocketflag/go-sdk"
+)
+
+// Client is an interface for the RocketFlag client.
+type Client interface {
+	GetFlag(flag string, user rocketflag.UserContext) (*rocketflag.FlagStatus, error)
+}
+
+type Provider struct {
+	client Client
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(client Client, opts ...ProviderOption) *Provider {
+	provider := &Provider{
+		client: client,
+	}
+	for _, opt := range opts {
+		opt(provider)
+	}
+	return provider
+
+}
+
+// Metadata returns value of Metadata (name of current service, exposed to openfeature sdk)
+func (p *Provider) Metadata() openfeature.Metadata {
+	return openfeature.Metadata{
+		Name: "RocketFlag",
+	}
+}
+
+// Hooks are not supported by RocketFlag.
+func (p *Provider) Hooks() []openfeature.Hook {
+	return []openfeature.Hook{}
+}
+
+func (p *Provider) BooleanEvaluation(ctx context.Context, flag string, defaultValue bool, evalCtx openfeature.FlattenedContext) openfeature.BoolResolutionDetail {
+	rocketflagUserContext := rocketflag.UserContext{}
+
+	// Validate "targetingKey" exists and is not nil
+	// Safely assert the underlying type is a string
+	// If it is a non-empty string, assign it to "cohort", which can then be passed to the rocketflag service
+	if targetingKey, ok := evalCtx["targetingKey"]; ok && targetingKey != nil {
+		if keyAsString, isString := targetingKey.(string); isString && keyAsString != "" {
+			rocketflagUserContext["cohort"] = keyAsString
+		}
+	}
```

Comment:
```suggestion
	if targetingKey, ok := flatCtx[openfeature.TargetingKey].(string); ok {
		rocketflagUserContext["cohort"] = targetingKey
	}
```
This can be simplified quite a bit. The targetingKey will not be in the flatCtx if it is an empty string and you can use the openfeature.TargetingKey const.

Reply:
Great insight, thanks.